### ISSUE TITLE
[Bug]: 11.0.2 release has broken Farming Bar

### ISSUE THREAD

**Issue by karpana** [nikkisaurus/FarmingBar#65]
### Interface Version

11.0.2

### AddOn Version

3.10.1

### Have you replicated this error without any other addons loaded?

Yes

### What steps did you take leading up to this issue?

start the game with the addon loaded.
farming bars don't show. and errors presented in bugsack

### Did you encounter any Lua errors related to this issue? If so, please provide them here.

```lua
108x FarmingBar/Modules/DataStore.lua:77: attempt to call global 'IsAddOnLoaded' (a nil value)
[string "@FarmingBar/Modules/DataStore.lua"]:77: in function `GetMissingDataStoreModules'
[string "@FarmingBar/Modules/Counts.lua"]:52: in function `GetTrackerCount'
[string "@FarmingBar/Modules/Counts.lua"]:19: in function `GetObjectiveWidgetCount'
[string "@FarmingBar/Widgets/Button.lua"]:417: in function `SetCount'
[string "@FarmingBar/Widgets/Button.lua"]:637: in function `UpdateAttributes'
[string "@FarmingBar/Widgets/Button.lua"]:630: in function `Update'
[string "@FarmingBar/Widgets/Button.lua"]:504: in function `SetID'
[string "@FarmingBar/Widgets/Bar.lua"]:115: in function `DrawButtons'
[string "@FarmingBar/Widgets/Bar.lua"]:287: in function `SetID'
[string "@FarmingBar/Modules/Bars.lua"]:93: in function `InitializeBars'
[string "@FarmingBar/FarmingBar.lua"]:20: in function <FarmingBar/FarmingBar.lua:14>
[string "=[C]"]: ?
...
[string "@Blizzard_UIParent/Mainline/UIParent.lua"]:627: in function `MajorFactions_LoadUI'
[string "@Blizzard_ExpansionLandingPage/Blizzard_DragonflightLandingPage.lua"]:103: in function `SetUpMajorFactionList'
[string "@Blizzard_ExpansionLandingPage/Blizzard_DragonflightLandingPage.lua"]:96: in function `RefreshMajorFactionList'
[string "@Blizzard_ExpansionLandingPage/Blizzard_DragonflightLandingPage.lua"]:91: in function `RefreshOverlay'
[string "@Blizzard_ExpansionLandingPage/Blizzard_DragonflightLandingPage.lua"]:83: in function <...sionLandingPage/Blizzard_DragonflightLandingPage.lua:79>
[string "=[C]"]: ?
[string "=[C]"]: in function `CreateFrame'
[string "@Blizzard_ExpansionLandingPage/Blizzard_DragonflightLandingPage.lua"]:49: in function `CreateOverlay'
[string "@Blizzard_ExpansionLandingPage/Blizzard_ExpansionLandingPage.lua"]:78: in function `RefreshExpansionOverlay'
[string "@Blizzard_ExpansionLandingPage/Blizzard_ExpansionLandingPage.lua"]:37: in function <...pansionLandingPage/Blizzard_ExpansionLandingPage.lua:35>

Locals:
self = <table> {
 defaults = <table> {
 }
 ObjectiveFrame = FarmingBarObjectiveFrame {
 }
 options = <table> {
 }
 bars = <table> {
 }
 FileData = <table> {
 }
 db = <table> {
 }
 CONST = <table> {
 }
 FileDataClassic = <table> {
 }
 status = <table> {
 }
 templates = <table> {
 }
 editor = <table> {
 }
 lists = <table> {
 }
 anchorPoints = <table> {
 }
}
(*temporary) = nil
(*temporary) = "DataStore"
(*temporary) = "attempt to call global 'IsAddOnLoaded' (a nil value)"
missing = <table> {
}
private = <table> {
 defaults = <table> {
 }
 ObjectiveFrame = FarmingBarObjectiveFrame {
 }
 options = <table> {
 }
 bars = <table> {
 }
 FileData = <table> {
 }
 db = <table> {
 }
 CONST = <table> {
 }
 FileDataClassic = <table> {
 }
 status = <table> {
 }
 templates = <table> {
 }
 editor = <table> {
 }
 lists = <table> {
 }
 anchorPoints = <table> {
 }
}

```

### Please supply a copy of your saved variables. You may need to save the file as a .txt file to upload here.

```lua
FarmingBarDB = {
["global"] = {
["version"] = 7,
["settings"] = {
["tooltips"] = {
["useGameTooltip"] = true,
["showDetails"] = true,
},
},
},
["profileKeys"] = {
["Carthusian - Arygos"] = "Default",
["Astronomer - Ysera"] = "Default",
["Karpana - Thorium Brotherhood"] = "Default",
["Laadora - Ysera"] = "Default",
["Varlooch - Ysera"] = "Default",
["Silanah - Arygos"] = "Default",
["Plutocrat - Ysera"] = "Default",
["Salliistari - Stormrage"] = "Default",
["Tibal - Ysera"] = "Default",
["Zardaukar - Stormrage"] = "Default",
["Leandro - Ysera"] = "Default",
["Rhyu - Ysera"] = "Default",
["Gryllen - Ysera"] = "Default",
["Physicist - Llane"] = "Default",
["Karpana - Doomhammer"] = "Default",
["Parkana - Ysera"] = "Default",
["Mandevyne - Arygos"] = "Default",
["Karpana - Garrosh"] = "Default",
["Parkana - Arygos"] = "Default",
["Venhold - Ysera"] = "Default",
["Carthusian - Ysera"] = "Default",
["Fuzy - Llane"] = "Default",
["Rythok - Ysera"] = "Default",
["Febryl - Ysera"] = "Default",
["Karpana - Ysera"] = "Default",
["Karpana - Aman'Thul"] = "Default",
["Aantheaoria - Illidan"] = "Default",
["Emurlahn - Ysera"] = "Default",
["Vqshonon - Area 52"] = "Default",
["Bleys - Ysera"] = "Default",
["Ghostwheel - Ysera"] = "Default",
},
["profiles"] = {
["Default"] = {
["bars"] = {
{
["buttonPadding"] = 0,
["point"] = {
"TOPRIGHT",
nil,
"TOPRIGHT",
0,
-332.380615234375,
},
["scale"] = 0.65,
["buttonsPerAxis"] = 14,
["buttonGrowth"] = "COL",
["skin"] = "FarmingBar_Minimal",
["hideInCombat"] = true,
["numButtons"] = 56,
["buttons"] = {
[55] = {
["onUse"] = {
["macrotext"] = "",
["type"] = "ITEM",
["itemID"] = 211515,
},
["trackers"] = {
{
["type"] = "ITEM",
["objective"] = 1,
["includeAllFactions"] = false,
["id"] = 211515,
["includeBank"] = false,
["name"] = "Splintered Spark of Awakening",
["includeGuildBank"] = {
},
["includeAlts"] = false,
["altIDs"] = {
},
},
},
["objective"] = 2,
["mute"] = false,
["icon"] = {
["id"] = 4638425,
["type"] = "AUTO",
},
["condition"] = {
["func"] = "function(trackers, GetTrackerCount)\n    -- This is the structure of the tracker table:\n    --trackers = {\n    --    [1] = {\n    --        type = \"ITEM\", -- \"CURRENCY\"\n    --        id = 0000,\n    --        objective = 1,\n    --        includeAllFactions = false,\n    --        includeAlts = false,\n    --        includeBank = false,\n    --        includeGuildBank = {\n    --             [\"GuildKey\"] = true\n    --        },\n    --        altIDs = {\n    --            {\n    --                type = \"ITEM\", -- \"CURRENCY\"\n    --                name = \"\",\n    --                id = 0000,\n    --                multiplier = 1,\n    --            }\n    --        },\n    --    },\n    --}\n    \n    -- NOTE: make sure the first argument of GetTrackerCount is nil\n    -- This function is not necessary, but available if your custom function is simple and doesn't change the way each tracker is calculated.\n    --local count =  GetTrackerCount(_, trackers[1])\n    \n    return GetTrackerCount(_, trackers[1])\nend",
["type"] = "ALL",
},
["title"] = "Splintered Spark of Awakening",
},
[54] = {
["onUse"] = {
["macrotext"] = "",
["type"] = "ITEM",
["itemID"] = 94111,
},
["trackers"] = {
{
["type"] = "ITEM",
["objective"] = 1,
["includeAllFactions"] = false,
["id"] = 94111,
["includeBank"] = true,
["name"] = "Lightning Steel Ingot",
["includeGuildBank"] = {
},
["includeAlts"] = false,
["altIDs"] = {
},
},
},
["objective"] = 100,
["mute"] = false,
["condition"] = {
["func"] = "function(trackers, GetTrackerCount)\n    -- This is the structure of the tracker table:\n    --trackers = {\n    --    [1] = {\n    --        type = \"ITEM\", -- \"CURRENCY\"\n    --        id = 0000,\n    --        objective = 1,\n    --        includeAllFactions = false,\n    --        includeAlts = false,\n    --        includeBank = false,\n    --        includeGuildBank = {\n    --             [\"GuildKey\"] = true\n    --        },\n    --        altIDs = {\n    --            {\n    --                type = \"ITEM\", -- \"CURRENCY\"\n    --                name = \"\",\n    --                id = 0000,\n    --                multiplier = 1,\n    --            }\n    --        },\n    --    },\n    --}\n    \n    -- NOTE: make sure the first argument of GetTrackerCount is nil\n    -- This function is not necessary, but available if your custom function is simple and doesn't change the way each tracker is calculated.\n    --local count =  GetTrackerCount(_, trackers[1])\n    \n    return GetTrackerCount(_, trackers[1])\nend",
["type"] = "ALL",
},
["icon"] = {
["id"] = 576649,
["type"] = "AUTO",
},
["title"] = "Lightning Steel Ingot",
},
[56] = {
["onUse"] = {
["macrotext"] = "",
["type"] = "ITEM",
["itemID"] = 211516,
},
["trackers"] = {
{
["type"] = "ITEM",
["objective"] = 1,
["includeAllFactions"] = false,
["id"] = 211516,
["includeBank"] = false,
["name"] = "Spark of Awakening",
["includeGuildBank"] = {
},
["includeAlts"] = false,
["altIDs"] = {
},
},
},
["objective"] = 1,
["mute"] = false,
["condition"] = {
["func"] = "function(trackers, GetTrackerCount)\n    -- This is the structure of the tracker table:\n    --trackers = {\n    --    [1] = {\n    --        type = \"ITEM\", -- \"CURRENCY\"\n    --        id = 0000,\n    --        objective = 1,\n    --        includeAllFactions = false,\n    --        includeAlts = false,\n    --        includeBank = false,\n    --        includeGuildBank = {\n    --             [\"GuildKey\"] = true\n    --        },\n    --        altIDs = {\n    --            {\n    --                type = \"ITEM\", -- \"CURRENCY\"\n    --                name = \"\",\n    --                id = 0000,\n    --                multiplier = 1,\n    --            }\n    --        },\n    --    },\n    --}\n    \n    -- NOTE: make sure the first argument of GetTrackerCount is nil\n    -- This function is not necessary, but available if your custom function is simple and doesn't change the way each tracker is calculated.\n    --local count =  GetTrackerCount(_, trackers[1])\n    \n    return GetTrackerCount(_, trackers[1])\nend",
["type"] = "ALL",
},
["icon"] = {
["id"] = 4693222,
["type"] = "AUTO",
},
["title"] = "Spark of Awakening",
},
},
["movable"] = false,
["fontstrings"] = {
["Count"] = {
["x"] = 0,
["y"] = 0,
},
["Objective"] = {
["x"] = 0,
["y"] = 0,
},
},
},
},
["style"] = {
["buttons"] = {
["padding"] = 0,
},
},
},
},
}
FarmingBarDevDB = nil
```

**Comment by Mythris29** [comment]
I'm getting this same error as well.

**Comment by karpana** [comment]
Looking at what some other addons have done to resolve this error ....

I opened up the `FarmingBar\Modules\DataStore.lua` file and added this at line 77

`local IsAddOnLoaded = C_AddOns.IsAddOnLoaded or IsAddOnLoaded`

the result of the code change is this:
```lua
    wipe(missing)

    local IsAddOnLoaded = C_AddOns.IsAddOnLoaded or IsAddOnLoaded
    
    if not IsAddOnLoaded("DataStore") then
```

addon works like a charm.

**Comment by nikkisaurus** [comment]
Should be fixed in [v3.11.0](https://github.com/nikkisaurus/FarmingBar/releases/tag/v3.11.0). Feel free to reopen if there are further issues.
